migration-guides/0.14-Migration-Guide.md

```diff
@@ -0,0 +1,597 @@
+Data.Argonaut.Encode.Generic.Rep -> Data.Argonaut.Encode.Generic
+Data.Argonaut.Types.Generic.Rep  -> Data.Argonaut.Types.Generic
+```
+
+#### The `globals` library has been deprecated
+
+The `globals` library provided various functions and values that are in the global scope in JavaScript. The library has been deprecated in particular because PureScript supports multiple backends besides JavaScript, and the relevant functions have been moved or reimplemented in other libraries.
+
+Specifically:
+
+* Number values and operations like `isFinite`, parsing, and formatting have been moved to the `numbers` package, which itself has moved into the core organization.
+* Integer parsing has been removed, as it is already included in the `integers` package.
+* Encoding and decoding URIs and URI components using JavaScript's builtin `encodeURIComponent` and `decodeURIComponent` have been moved to the new `js-uri` package.
+* Unsafe global functions (i.e. those which lived in `Global.Unsafe`) have been removed altogether.
+
+**To fix:**
+
+You should remove the `globals` dependency from your library or application.
+
+If you used any `Number`-related code from the `globals` package, then you should add a new dependency on `numbers` and rename your imports:
+
+```diff
+- import Global (isNan, nan, isFinite, infinity)
++ import Data.Number (isNan, nan, isFinite, infinity)
+ 
+- import Global (toFixed, toPrecision, toExponential)
++ import Data.Number.Format (toStringWith, fixed, precision, exponential)
+
+  -- toFixed, toPrecision, and toExponential can be replaced as follows:
+- x = toFixed someInt 4.0 
++ x = toStringWith (fixed someInt) 4.0
+ 
+- y = toPrecision someInt 4.0
++ y = toStringWith (precision someInt) 4.0
+ 
+- z = toExponential intArg 4.0
++ z = toStringWith (exponential someInt) 4.0
+```
+
+If you used `readInt`, then you should add a dependency on `integers` and either use `Data.Int.fromString` if you were parsing a base-10 integer, or use `Data.Int.fromStringAs` given a `Data.Int.Radix` if you were parsing a number with another base:
+
+```diff
+- import Global (readInt)
++ import Data.Int (binary, fromStringAs)
+
+  fromBinaryString :: String -> Maybe Int
+- fromBinaryString = readInt 2
++ fromBinaryString = fromStringAs binary
+```
+
+If you used URI encoding and decoding functions, then you should add a dependency on `js-uri` and update your imports:
+
+```diff
+- import Global (decodeURI, decodeURIComponent, encodeURI, encodeURIComponent)
++ import JSURI (decodeURI, decodeURIComponent, encodeURI, encodeURIComponent)
+```
+
+If you used various unsafe functions exposed by the `globals` package, you can in most cases use functions from the `partials` package (like `unsafePartial`) combined with partial functions like `Data.Maybe.fromJust` to produce an unsafe function from a safe one.
+
+#### The dependency graph for `functors` and related packages have changed in order to unify newtypes among the libraries.
```

Comment:
This should probably be folded into the earlier section about library restructuring

Reply:
Never mind